docs/articles/immutability-in-js.md

```diff
@@ -0,0 +1,108 @@
+
+### Array.concat
+
+Now let's do it with an array. We have an array and we want to add two new elements in an immutable way.
+
+https://gist.github.com/frinyvonnick/461d4dbad6bff965a60b734f86619faa
+
+`Array.concat` doesn't mutate our array in contrary of `Array.push` an return a new array without using trick like we did with `Object.assign`.
+
+https://gist.github.com/frinyvonnick/3959e7972688a41055671bceb6228056
+
+This method is flexible. It takes as many elements as you want and they can be values or arrays.
+
+### Object.freeze
+
+`Object.freeze` isn't really familiar. It let you make an object immutable ! It prevents every type of mutation (creation, modification, deletion) included using setter.
+
+https://gist.github.com/frinyvonnick/d1dd1cbf2924b2428b07713d1ad2adf7
+
+We will try to delete name property after object has been frozen.
+
+https://gist.github.com/frinyvonnick/acc86d1c2fe14c23cd6363946cc253da
+
+Reallocation isn't necessary since the object passed as parameter is made immutable by `Object.freeze`. This methode has two modes available:
+
+- In non-strict mode it doesn't apply mutations
+- In strict it throws a `TypeError` if you try to apply mutations on a frozen object
+
+Be carefull, it isn't recursive. So our property commonNames isn't immutable.
+
+### Spread operator
+
+The spread operator syntax have been introduced in ES2015 for arrays and ES2018 for objects. It let's you copy all properties of an object in a new object literal.
+
+https://gist.github.com/frinyvonnick/61908a0e1bdcd97ba965570796c0c665
+
+For arrays, it copies all values of an array to a new array literal.
+
+https://gist.github.com/frinyvonnick/7ec4cd039fd79f16864546c51b18ab95
+
+Easily readable and unified between arrays and objects. It replaces nicely assign and concat. It is possible to spread multiple arrays and object in a same literal.
+
+## Why use immutability ?
+
+You found out how to make objects and arrays immutable with JavaScript but we didn't tell why using immutability is so necessary nowadays. As developper, we always search for a way to write more maintainable and readable code. Some paradigms as Functionnal Programming are focusing on this.
+
+> Functional programming’s goal is to allow us to think less and write more descriptive code.
+>
+> -- <cite>[Alexander Kondov](https://hackernoon.com/functional-programming-paradigms-in-modern-javascript-immutability-4e9751ca005c)</cite>
+
+It has a declarative approach of programming that means you focus on describing what your program must accomplish rather than how it should does it. It brings more meaning to your code so the next developper will understand it more easily. Functional programming brings alongside other concepts that helps to reach this goal, such as Immutability.
+
+### What are the benefits ?
+
+It sounds like a hype term to you ? Immutability answer to some problems we encounter everyday while programming:
```

Comment:
`to you?`
`Immutability addresses some problems we encounter everyday while programming:`